### ISSUE TITLE
Jittering when using FlatList instead of ScrollView

### ISSUE THREAD

**Issue by samw2k00** [sonaye/react-native-collapsible-header#3]
Hi Sonaye,

Thank you for this awesome collapsible. I recalled you wrote one version using FlatList instead of ScrollView.
Im implementing exactly the same thing but i find the scrolling having huge issue with the collapsible header, when scrolling through large item that get render in and out base on what is on screen (Flatlist behaviour) can cause the header to drop down and bounce back up. This mean that when you scroll very fast, you will see the header bounce down and up many times causing bad jittering (like low frame rate). By using scroll view, i have no problem with this when using scroll view but when a large amount of items required to be render, scroll view is not efficient.

Note, this behaviour occur when the render of each Item is slighly different to other, because the FlatList need to re-calculate the height when a new item come into screen.

I noticed that one of your commit, you move from FlatList to scroll view to fix this behaviour. 
https://github.com/sonaye/react-native-collapsible-header/commit/035b501e27ce59354dceae7fdf94f64a801aae36

https://github.com/samw2k00/react-native-collapsible-header

Im really interested in fixing the FlatList and would love to contribute where i can

**Comment by sonaye** [comment]
Hey!

Indeed I switched from FlatList to ScrollView at some point to improve the behavior and the overall feeling of the component which I felt was lacking, ScrollView makes tapping into specific configurations much easier, that's why. One thing to note as well, FlatList itself is essentially built on top of ScrollView.

As far as I know, you can use the new API (built on top of ScrollView) and render a FlatList using the new `renderContent` prop, I remember testing this specific case before deprecating FlatList as the base component, and it worked like charm. Can you try implementing your concept with the new API? with a FlatList as the content. Let me know about the results.

```javascript
<Collapsible
  renderHeader={<Header />}
  renderContent={<FlatList />}
/>
```

It would help as well if you can provide a gif of what's going on in case my suggestion to use the new API doesn't work. I will check on the repo you provided as soon as I have the chance.

**Comment by samw2k00** [comment]
Hey Sonaye,

Thank you for the response, i tested out the FlatList inside the render Content, indeed the Collapsible accept the FlatList but the behaviour is slight change now.
- The collapsible header does collapse, but because i added the FlatList in which itself have it own panResponder, which mean that the scrolling the content back up does not initiate the collapsible to come back down. I have to scroll right to the top (top of the scroll view) of itself to get the collapsible header back down.
- I noticed the slowness when having a huge list of content in the FlatList.
- The tap on the status bar no longer work, ( this usually is the function where it automatically scroll to the top of the list)

Im working on another version using panResponder instead, im almost there but i got a minor issue, i think its the offset but im not 100% sure yet.
The problem is when  i start scrolling, the header tap drop back down to the original position

Here is a work in progress code.

https://github.com/samw2k00/react-native-collapsible-header/blob/master/CollapsibleHeaderWithPanResponder.js

GIF -->  https://dl.dropboxusercontent.com/u/580042/Jul-17-2017%2000-13-16.gif

If you have time to have a look would we super awesome,

**Comment by samw2k00** [comment]
Hey,

I managed to to get the PanResponder working. I think it may be a hack but its work for me.....

https://github.com/samw2k00/react-native-collapsible-header/blob/master/CollapsibleHeaderWithPanResponder.js

**Comment by sonaye** [comment]
> scrolling the content back up does not initiate the collapsible to come back down

I removed this behavior when I moved to ScrollView because it was extremely glitchy at the time, to the extent that I choose to completely ditch it over keeping a poorly implemented version of it.

This behavior can be achieved using these three props on the ScrollView
```
onScrollEndDrag
onMomentumScrollBegin
onMomentumScrollEnd
```

Take a look at the following links for more info.
- [Inspiration](https://medium.com/appandflow/react-native-collapsible-navbar-e51a049b560a)
- [Snack]( https://snack.expo.io/B1v5RS7ix)

> I noticed the slowness when having a huge list of content in the FlatList

This shouldn't happen, particularly with FlatList, always evaluate the performance and the animations on a real device and not on a simulator.

> The tap on the status bar no longer work

I actually never wrote a piece of code that handle this feature, very interesting idea and intuitive actually, not sure why it behaved that way before, perhaps it was treating the tap as a scroll of some sort.

> Im working on another version using panResponder instead

Adding panResponder to ScrollView -or FlatList- doesn't seem right to me for some reason, ScrollView offers props to accomplish that, we just need to find them. Glad it worked though, I will definitely go through your code.

Actually the main reason I moved from FlatList was https://github.com/sonaye/react-native-collapsible-header/issues/2, make sure you test on Android too. Let me know if it works.

I've added a couple of todos based on this issue https://github.com/sonaye/react-native-collapsible-header/issues/5 https://github.com/sonaye/react-native-collapsible-header/issues/6, they are worth looking into. I'll try to implement them without involving panResponder at first, if that didn't work, I might steal your some pieces of your implementation and add it to the component.

**Comment by sonaye** [comment]
Support for FlatList is now available in `0.0.13`.
